transport/response.hh

```diff
@@ -55,7 +55,8 @@ public:
     template<typename T>
     class placeholder;
 
-    response(int16_t stream, cql_binary_opcode opcode, const tracing::trace_state_ptr& tr_state_ptr)
+    response(int16_t stream, cql_binary_opcode opcode, const tracing::trace_state_ptr& tr_state_ptr,
+             std::optional<std::map<sstring, bytes>> custom_payload = std::nullopt)
```

Comment:
Small nit: Now that I have thought more about it, two constructors:
1. `response(int16_t stream, cql_binary_opcode opcode, const tracing::trace_state_ptr& tr_state_ptr)`
2. `response(int16_t stream, cql_binary_opcode opcode, const tracing::trace_state_ptr& tr_state_ptr, std::map<sstring, bytes> custom_payload)`

would probably be a better solution.